source/hash_table.c

```diff
@@ -556,12 +566,31 @@ int aws_hash_table_foreach(struct aws_hash_table *map,
 
 void aws_hash_table_clear(struct aws_hash_table *map) {
     struct hash_table_state *state = map->pImpl;
-    if (state->destroy_fn) {
+    if (state->destroy_key_fn) {
```

Comment:
There's a lot of duplicate code in this loop. I'd do an `if (state->destroy_key_fn || state->destroy_value_fn)` at the top level and test for each individual one on the inner loop. Yes, it's very slightly slower, but in practice the call to whichever destructor exists will be much higher overhead than the inner loop guards.

Reply:
I agree that making this change can make for slightly simpler code, but in my opinion the efficiency gain is important here. Hash tables can get very large. Doing redundant checks of the same flag every iteration through is wasteful. Since this is a general purpose library that anyone who is using hash tables of any size while working with AWS in C might use, it makes sense to me to do a best possible implementation rather than sacrifice quality for a little bit of extra clarity.

Furthermore, this function isn't that long or hard to read as is. I got rid of the blank lines, so now the entire function is under 30 lines. I also looked it over with another coworker and on her advice added a few extra comments to improve the clarity.